src/pages/memos/[functionId]/MemosForm.js

```diff
@@ -0,0 +1,531 @@
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { SystemFunction } from 'src/resources/SystemFunction'
+import { useDocumentType } from 'src/hooks/documentReferenceBehaviors'
+
+export default function MemosForm({ labels, access, recordId, functionId, getEndpoint }) {
+  const { documentType, maxAccess, changeDT } = useDocumentType({
+    functionId: functionId,
+    access: access,
+    enabled: !recordId
+  })
+
+  const [initialVatPct, setInitialVatPct] = useState('')
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.FiMemo.qry
+  })
+
+  // const getSetEndpoint = functionId => {
+  //   switch (functionId) {
+  //     case SystemFunction.CreditNote:
+  //       return FinancialRepository.CreditNote.set
+  //     case SystemFunction.DebitNote:
+  //       return FinancialRepository.DebitNote.set
+  //     case SystemFunction.ServiceBill:
+  //       return FinancialRepository.ServiceBillReceived.set
+  //     case SystemFunction.ServiceInvoice:
+  //       return FinancialRepository.ServiceInvoice.set
+  //     default:
+  //       return null
+  //   }
+  // }
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      dtId: documentType?.dtId || null,
+      reference: '',
+      date: new Date(),
+      plantId: '',
+      currencyId: '',
+      status: '',
+      accountId: '',
+      amount: '',
+      baseAmount: '',
+      functionId: functionId,
+      vatPct: '',
+      exRate: '',
+      rateCalcMethod: '',
+      subtotal: '',
+      notes: '',
+      vatAmount: '',
+      isSubjectToVAT: false
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      amount: yup.number().required(' '),
+      currencyId: yup.string().required(' '),
+      accountId: yup.string().required(' '),
+      subtotal: yup.number().required(' '),
+      date: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+      if (!recordId) {
+        obj.baseAmount = obj.amount
+        obj.status = 1
+        obj.rateCalcMethod = 1
+        obj.exRate = 1
+      }
+
+      const response = await postRequest({
+        extension: getEndpoint(parseInt(formik.values.functionId)).set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!recordId) {
+        toast.success('Record Added Successfully')
+        formik.setValues({
+          ...obj,
+          baseAmount: obj.amount,
+
+          recordId: response.recordId
+        })
+      } else {
+        toast.success('Record Edited Successfully')
+      }
+
+      try {
```

Comment:
let the try block includes the first post request also

Reply:
correct. please fix.